pkg/didmethod/peer/validate.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	"regexp"
+	"unicode"
+
+	errors "golang.org/x/xerrors"
+)
+
+func validate(did string) error {
```

Comment:
In case of Peer DIDs, the validation should be done against the genesis version of the document.

Quoting the [spec](https://openssi.github.io/peer-did-method-spec/index.html#create-register)

> The inviter must verify that the received DID was generated according to the namestring generation method from the stored variant of the received genesis DID Doc

Reply:
Interesting - I previously thought [here](https://github.com/hyperledger/aries-framework-go/pull/56#issuecomment-517830852) the Recipient of a peer DID should not attempt this validation. Turns out that by enforcing this validation we help prevent MiTM attacks as explained [here](https://openssi.github.io/peer-did-method-spec/index.html#proof-of-control).

The implication is that a party that wants to reuse the same peer DID for another connection must replay all changes back to the new recipient.

Reply:
@talwinder50 any updates on this topic and this PR?

Reply:
@troyronda pushing patch set in next 15 min